response.go

```diff
@@ -135,19 +141,21 @@ func unpackResponseBody(buf []byte, rspObj interface{}) error {
 			return perrors.WithStack(err)
 		}
 		if e, ok := expt.(error); ok {
-			return e
+			response.Exception = e
+			return nil
 		}
-		return perrors.Errorf("got exception: %+v", expt)
+		response.Exception = perrors.Errorf("got exception: %+v", expt)
+		return nil
 
 	case RESPONSE_VALUE, RESPONSE_VALUE_WITH_ATTACHMENTS:
 		rsp, err := decoder.Decode()
 		if err != nil {
 			return perrors.WithStack(err)
 		}
-		return perrors.WithStack(ReflectResponse(rsp, rspObj))
+		return perrors.WithStack(ReflectResponse(rsp, response.RspObj))
 
 	case RESPONSE_NULL_VALUE, RESPONSE_NULL_VALUE_WITH_ATTACHMENTS:
-		return perrors.New("Received null")
+		return nil
```

Comment:
Why use nil instead of a error?

Reply:
`RESPONSE_NULL_VALUE` and `RESPONSE_NULL_VALUE_WITH_ATTACHMENTS` are normal.
They should not be an error, and that is similar to `RESPONSE_WITH_EXCEPTION`.

Reply:
got it.